WordPress/gutenberg
# Add reset button to ColorGradientSettingsDropdown

## What?


Follow-up of https://github.com/WordPress/gutenberg/pull/67116#pullrequestreview-2469794583
Fixing https://github.com/WordPress/gutenberg/issues/41866

Add an inline reset button to the ColorGradientSettingsDropdown, shown on hover/focus, if there's a value. This will complete the fix started in https://github.com/WordPress/gutenberg/pull/67116, as there are different blocks that are using the ColorGradientSettingsDropdown (navigation, cover, social links)

Also, for the original implementation toggling the control, if it's open to avoid having multiple controls opened at the same time.

## Why?


To make it easier to reset a color selection, reduce the number of clicks as it doesn't require opening the color options panel.

## How?


Adding a button if there's a color value, to reset it. I'm leveraging the styles introduced in https://github.com/WordPress/gutenberg/pull/67116 for the reset button.

## Testing Instructions

1. Create a page.
2. Add a social icons block.
3. Add some icons
4. Confirm that if some colors are selected for the block, you see a hover button to "reset" the color.
5. Repeat the process with the keyboard navigation and confirm that the focus is not lost.

## Screenshots or screencast 

https://github.com/user-attachments/assets/f408224e-a411-419a-bb9b-88add3db5a68
